### DISCUSSION TITLE
Vscode Crashes on Niri, works fine on Gnome

### DISCUSSION THREAD

**Discussion by binamralamsal** [YaLTeR/niri#343] [General]
I am using `niri 0.1.5 (6a80078)`. And I tried installing Vscode using DNF, Flatpak, Snaps, and tried bunch of solutions that I found online but no help. Vscode doesn't open or crashes. It works fine when I open on Gnome.

Specifications:
OS: Fedora Linux 40 (Workstation Edition) x86_64
Host: Aspire A515-51G V1.19
Kernel: 6.8.5-301.fc40.x86_64
CPU: Intel i5-7200U (4) @ 3.100GHz
GPU: NVIDIA GeForce 940MX
GPU: Intel HD Graphics 620
Memory: 8gb
When I open Vscode using "code" command then nothing happens if installed using DNF.
but when using `flatpak run com.visualstudio.code`, I get a warning:
```
[2 zypak-helper] Wait found events, but sd-event found none
Warning: 'ms-enable-electron-run-as-node' is not in the list of known options, but still passed to Electron/Chromium.
```

I am not sure if this issue is because of Niri or something else, that's why I am creating a discussion.

**Comment by YaLTeR** [comment]
You need to pass it flags to run using the Wayland backend (rather than the default X11). On Flatpak, you also need to make sure you enable the Wayland socket permission.

`code --ozone-platform-hint=auto`

**Comment by binamralamsal** [comment]
Thank you so much, it worked
